main.go

```diff
@@ -2,12 +2,79 @@ package main
 
 import (
 	"fmt"
+	"log"
+	"os"
+
+	"github.com/urfave/cli/v2"
 )
 
+type Config struct {
+	dropletName    string
+	accessToken    string
+	sshFingerprint string
+	url            string
+	branch         string
+	rancherVersion string
+}
+
 func main() {
-	fmt.Println("Provisioning Digital Ocean Droplet...")
-	digitalOceanId, _ := CreateDroplet(Prompt())
+	var config Config
+
+	app := &cli.App{
+		Name:  "do",
+		Usage: "Quickly provision Rancher setups on Digital Ocean",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "droplet-name",
+				Usage:       "Name for your Droplet",
+				Required:    true,
+				Destination: &config.dropletName,
+			},
+			&cli.StringFlag{
+				Name:        "access-token",
+				Usage:       "Digital Ocean personal access token",
+				Required:    true,
+				Destination: &config.accessToken,
+			},
+			&cli.StringFlag{
+				Name:        "ssh-fingerprint",
+				Usage:       "Fingerprint for SSH Public Key",
+				Required:    true,
+				Destination: &config.sshFingerprint,
+			},
+			&cli.StringFlag{
+				Name:        "url",
+				Usage:       "Github url to provision",
+				DefaultText: "https://github.com/rancher/dashboard.git",
+				Destination: &config.url,
+			},
+			&cli.StringFlag{
+				Name:        "branch",
+				Usage:       "Git branch to target",
+				DefaultText: "master",
+				Destination: &config.branch,
+			},
+			&cli.StringFlag{
+				Name:        "rancher-version",
+				Usage:       "Target version of Rancher",
+				DefaultText: "v2.6-head",
```

Comment:
I don't think the default text is being respected - I tried to create a droplet while omitting this version and could see in the Dockerfile that the version field is empty. If the argument is provided though, it will be added to the Dockerfile.

Reply:
Good catch on this one. Created issue #9 so we can merge and take a closer look at this later 😁